samples/general/mobile_nerf_rayquery/mobile_nerf_rayquery.cpp

```diff
@@ -0,0 +1,1395 @@
+	attachments[0].initialLayout  = VK_IMAGE_LAYOUT_UNDEFINED;
+	attachments[0].finalLayout    = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+	// Swapchain  attachment
+	attachments[1].format         = get_render_context().get_swapchain().get_format();
+	attachments[1].samples        = VK_SAMPLE_COUNT_1_BIT;
+	attachments[1].loadOp         = VK_ATTACHMENT_LOAD_OP_CLEAR;
+	attachments[1].storeOp        = VK_ATTACHMENT_STORE_OP_STORE;
+	attachments[1].stencilLoadOp  = VK_ATTACHMENT_LOAD_OP_DONT_CARE;
+	attachments[1].stencilStoreOp = VK_ATTACHMENT_STORE_OP_DONT_CARE;
+	attachments[1].initialLayout  = VK_IMAGE_LAYOUT_UNDEFINED;
+	attachments[1].finalLayout    = VK_IMAGE_LAYOUT_PRESENT_SRC_KHR;
+
+	VkAttachmentReference depth_reference = {};
+	depth_reference.attachment            = 0;
+	depth_reference.layout                = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+
+	VkAttachmentReference swapchain_reference = {};
+	swapchain_reference.attachment            = 1;
+	swapchain_reference.layout                = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+
+	VkSubpassDescription subpass    = {};
+	subpass.pipelineBindPoint       = VK_PIPELINE_BIND_POINT_GRAPHICS;
+	subpass.colorAttachmentCount    = 1;
+	subpass.pColorAttachments       = &swapchain_reference;
+	subpass.pDepthStencilAttachment = &depth_reference;
+	subpass.inputAttachmentCount    = 0;
+	subpass.pInputAttachments       = nullptr;
+	subpass.preserveAttachmentCount = 0;
+	subpass.pPreserveAttachments    = nullptr;
+	subpass.pResolveAttachments     = nullptr;
+
+	VkRenderPassCreateInfo render_pass_create_info = {};
+	render_pass_create_info.sType                  = VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO;
+	render_pass_create_info.attachmentCount        = static_cast<uint32_t>(attachments.size());
+	render_pass_create_info.pAttachments           = attachments.data();
+	render_pass_create_info.subpassCount           = 1;
+	render_pass_create_info.pSubpasses             = &subpass;
+
+	VK_CHECK(vkCreateRenderPass(get_device().get_handle(), &render_pass_create_info, nullptr, &render_pass_nerf));
+}
+
+void MobileNerfRayQuery::setup_framebuffers()
+{
+	// Delete existing frame buffers
+	if (framebuffers_nerf.size() > 0)
+	{
+		for (uint32_t i = 0; i < framebuffers_nerf.size(); i++)
+		{
+			if (framebuffers_nerf[i] != VK_NULL_HANDLE)
+			{
+				vkDestroyFramebuffer(get_device().get_handle(), framebuffers_nerf[i], nullptr);
+			}
+		}
+	}
+
+	std::vector<VkImageView> views;
+	views.resize(2);
+	views[0] = depth_stencil.view;
+
+	// Depth/Stencil attachment is the same for all frame buffers
+	VkFramebufferCreateInfo framebuffer_create_info = {};
+	framebuffer_create_info.sType                   = VK_STRUCTURE_TYPE_FRAMEBUFFER_CREATE_INFO;
+	framebuffer_create_info.pNext                   = NULL;
+	framebuffer_create_info.renderPass              = render_pass_nerf;
+	framebuffer_create_info.attachmentCount         = views.size();
```

Comment:
`static_cast<uint32_t>` missing?

Reply:
Added this and a few more that were missing. Thanks.